internal/core/data/valuedescriptor_test.go

```diff
@@ -311,13 +313,15 @@ func TestGetValueDescriptorsByTypeNotFound(t *testing.T) {
 
 func TestGetValueDescriptorsByTypeError(t *testing.T) {
 	reset()
+
 	myMock := &mocks.DBClient{}
 
 	myMock.On("ValueDescriptorsByType", mock.Anything).Return([]models.ValueDescriptor{}, fmt.Errorf("some error"))
 
 	dbClient = myMock
+	mdc = newMockDeviceClient()
```

Comment:
See my other comment regarding the global variable changes needed due to the removal of the `TestMain` test in `device_test.go`